### PR TITLE
Bump buildbot to 3.11.0

### PR DESCRIPTION
**Pull request by zorry**

### CODE REVIEW

File: dev-util/buildbot/buildbot-3.11.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+KEYWORDS="~amd64 ~arm64 ~amd64-linux ~x86-linux"
+
+IUSE="crypt docker examples irc test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	acct-user/buildbot
+	~dev-util/buildbot-worker-${PV}[${PYTHON_USEDEP}]
+	>=dev-python/alembic-1.6.0[${PYTHON_USEDEP}]
+	>=dev-python/autobahn-0.16.9[${PYTHON_USEDEP}]
```
Comment (eli-schwartz) [inline dev-util/buildbot/buildbot-3.11.0.ebuild:27]: Due to re-sorting in the same commit, changes are a *tiny* bit hard to follow.

autobahn is bumped to .9 here, but not upstream?

File: dev-util/buildbot/buildbot-3.11.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+	>=dev-python/croniter-1.3.0[${PYTHON_USEDEP}]
+	>=dev-python/msgpack-0.6.0[${PYTHON_USEDEP}]
```
Comment (eli-schwartz) [inline dev-util/buildbot/buildbot-3.11.0.ebuild:29]: Good catch, required since buildbot 3.5.0

File: dev-util/buildbot-worker/buildbot-worker-3.11.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+resulting directories are owned by the \"buildbot\" user and point
+\"${ROOT}/etc/conf.d/buildbot_worker.myinstance\" at the right location.
+The scripts can	run as a different user if desired."
+
+src_prepare() {
+	# Remove shipped windows start script
+	sed -e "/'buildbot_worker_windows_service=buildbot_worker.scripts.windows_service:HandleCommandLine',/d" \
+		-i setup.py || die
+	# applay remove py 2.7 patch
+	eapply "${WORKDIR}/buildbot-worker-${PV}-remove_py27.patch"
```
Comment (eli-schwartz) [inline dev-util/buildbot-worker/buildbot-worker-3.11.0.ebuild:52]: This is... pretty obscure, and not obvious that it actually consists of a backport from upstream git master. 1100 lines worth of patch, in fact, which is a bit on the difficult side to guarantee it will continually rebase.

... which disappointingly, despite promises that it would be in the next release, is not in fact in the next release after all. I've pinged upstream: https://github.com/buildbot/buildbot/pull/6997#issuecomment-1913958150

Comment (zorry) [reply dev-util/buildbot-worker/buildbot-worker-3.11.0.ebuild:52]: It was suppose to be in next release.  Thanks for pinging upstream. Can add more info in the ebuild and patch. Buildbot is the last package that depend on future in gentoo.

Comment (eli-schwartz) [reply dev-util/buildbot-worker/buildbot-worker-3.11.0.ebuild:52]: Yeah I've been working pretty hard to get future into a position to be treecleaned so I know all about it. :D

buildbot rejected my conservative patch to move python2 support from future to six, because they said they'd release this patch for fully dropping python2 in the next minor release... now they did not do so. Baffling and disappointing.
